packages/pwa-buildpack/lib/BuildBus/mapHooksToTargets.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Mapping Tapable hooks to the Buildpack facade of Targets.
+ * @module Buildpack/BuildBus
+ */
+const Tapable = require('tapable');
+
+/**
+ * The names of Tapable hooks, but without "Hook" on the end.
+ * We hide the "Hook" name a little bit because its name overlaps
+ * with React Hooks in a confusing way.
+ * @private
+ */
+const allowedTargetTypes = [
+    'Sync',
+    'SyncBail',
+    'SyncWaterfall',
+    'SyncLoop',
+    'AsyncParallel',
+    'AsyncParallelBail',
+    'AsyncSeries',
+    'AsyncSeriesBail',
+    'AsyncSeriesWaterfall'
+];
+
+/**
+ * Map of Tapable hook classes to the 'Hook'-free target names in
+ * `allowedTargetTypes`
+ * @private
+ */
+const VALID_TYPES = new Map();
+
+/**
+ * Dictionary of Tapable Hook classes to expose under these new names.
+ * @public
+ * @type {Object.<string,Tapable.Hook>}
+ */
+const types = {};
+
+// Populate the validator map and type dictionary
+for (const type of allowedTargetTypes) {
+    const HookConstructor = Tapable[type + 'Hook'];
+    types[type] = HookConstructor;
+    VALID_TYPES.set(HookConstructor, type);
+}
+
+/**
+ * Duck typing for async hooks
+ * @private
+ */
+const hasAsyncHookInterface = thing =>
```

Comment:
Is there a more descriptive name we can use than `thing`? :)

Reply:
Changed.